xlive/Blam/Engine/objects/object_types.cpp

```diff
@@ -14,8 +14,92 @@ object_type_definition* object_type_definition_get(e_object_type object_type)
 	return get_object_type_definitions()[object_type];
 }
 
+object_type_definition* get_game_object_type_definition(datum object_datum)
+{
+	return object_type_definition_get(object_get_fast_unsafe(object_datum)->object_identifier.get_type());
+}
+
 void __cdecl object_type_adjust_placement(object_placement_data* placement_data)
 {
 	INVOKE(0x1863EA, 0x170486, object_type_adjust_placement, placement_data);
 	return;
-}
\ No newline at end of file
+}
+
+bool object_type_new(datum object_index, object_placement_data* placement_data, bool* some_bool)
+{
+	const object_type_definition* object_type = get_game_object_type_definition(object_index);
+	bool object_created = true;
```

Comment:
Should this not be initiated as false Incase the object creation fails?

Reply:
It was initialized as true in the original function